services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -0,0 +1,51 @@
+package cromwell.pipeline.service
+import java.nio.file.Path
+
+import cromwell.pipeline.datastorage.dto.{ Project, ProjectFile, Version }
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GitLabProjectVersioning(httpClient: HttpClient)(
+  implicit executionContext: ExecutionContext
+) extends ProjectVersioning[VersioningException] {
+
+  lazy val URL: String = "https://gitlab.com/api/v4/"
+  lazy val NAMESPACE
+    : String = "AdminLogin%2F" //to access repo in GitLab use this combo: NAMESPACE + project.projectId.value
+  lazy val TOKEN: Map[String, String] = Map("PRIVATE-TOKEN" -> "") //access_token must be here
+
+  override def updateFile(project: Project, projectFile: ProjectFile): AsyncResult[String] = ???
+  override def updateFiles(project: Project, projectFiles: ProjectFiles): AsyncResult[List[String]] = ???
+
+  override def createRepository(project: Project): AsyncResult[Project] = {
+    def responseFuture = httpClient.post(URL + "projects", createRepositoryParams(project), TOKEN, "")
+    if (project.active)
```

Comment:
I think you should first check if the project is active and if not do post request

Reply:
That's a question. I think an active project means that a repository is already created. And on success, I change the repository status for active.